mindsdb/integrations/handlers/ckan_handler/ckan_handler.py

```diff
@@ -1,101 +1,246 @@
-from ckanapi import RemoteCKAN as rc
 import pandas as pd
-
-from mindsdb.integrations.libs.base import DatabaseHandler
-from mindsdb.integrations.libs.response import HandlerStatusResponse, HandlerResponse, RESPONSE_TYPE
-from mindsdb_sql.parser.ast.base import ASTNode
-from mindsdb_sql.render.sqlalchemy_render import SqlalchemyRender
+from ckanapi import RemoteCKAN
+from mindsdb.integrations.libs.api_handler import APIHandler, APITable
+from mindsdb.integrations.libs.response import (
+    HandlerResponse,
+    HandlerStatusResponse,
+    RESPONSE_TYPE,
+)
+from mindsdb_sql.parser import ast
 from mindsdb.utilities import log
+from mindsdb.integrations.utilities.sql_utils import extract_comparison_conditions
 
 logger = log.getLogger(__name__)
 
 
-class CkanHandler(DatabaseHandler):
-    """
-    This handler handles connection and consumption of the CKAN API.
-    """
+class PackageIDsTable(APITable):
+    def select(self, query: ast.Select) -> HandlerResponse:
+        conditions = extract_comparison_conditions(query.where) if query.where else []
+        limit = query.limit.value if query.limit else 1000
+
+        packages = self.list(conditions, limit)
+        return HandlerResponse(RESPONSE_TYPE.TABLE, packages)
+
+    def list(self, conditions=None, limit=1000):
```

Comment:
By default, this will always work with 1000 rows, right? if so, we may add in the README